store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -8189,6 +8221,54 @@ public void modifyContact(OperationContext octxt, int contactId, ParsedContact p
         }
         return newAddrs;
     }
+    
+    /**
+     * Returns a list of contacts don't exist in any contacts folders.
+     *
+     * @param octxt OperationContext object
+     * @param addrs Recipient Email address list
+     * @return listOfContactsToBeModified This list consist list of Map, email as key and Pair of ContactId and 
+     * DisplayName as Value
+     */
+    public List<Map<String, Pair<Integer, String>>> findContactsInEmailedContacts
+    (OperationContext octxt, Collection<Address> addrs) {
+        if (addrs.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<Map<String, Pair<Integer, String>>> list = new ArrayList<>();
+        Map<String, Pair<Integer, String>> map = new HashMap<>();
+
+        for (Address addr : addrs) {
+            if (addr instanceof javax.mail.internet.InternetAddress) {
+                javax.mail.internet.InternetAddress iaddr = (javax.mail.internet.InternetAddress) addr;
+                try {
+                    if (!Strings.isNullOrEmpty(iaddr.getAddress())) {
+                        ZimbraQueryResults results = index.search(octxt, "in:\"Emailed Contacts\" field[email]:"+iaddr.getAddress()+"",
+                                EnumSet.of(MailItem.Type.CONTACT), SortBy.NONE, 1);
+                        while (results.hasNext()) {
+                            ZimbraHit hit = results.getNext();
+                            if (hit instanceof ContactHit) {
+                                ContactHit contactHit = (ContactHit) hit;
+                                Contact contact =contactHit.getContact();
+                                Map<String, String> m = contact.getFields();
+                                String fullName = m.get("fullName");
+                                String firstName = m.get("firstName");
+                                String emailSentName = iaddr.getPersonal();
+                                if (fullName != null && firstName != null &&!emailSentName.equalsIgnoreCase(fullName) 
```

Comment:
* add a white space: `&& !emailSentName.equalsIgnoreCase(fullName)`
* does it need to be case insensitive?

Reply:
done